network/network.py

```diff
@@ -0,0 +1,135 @@
+import random
+import select
+import re
+import socket
+import subprocess
+
+import psh.tools.psh as psh
+from trunner.ctx import TestContext
+from trunner.dut import Dut
+from trunner.types import TestResult, Status
+
+
+def host_interface():
+    out = subprocess.run(
+        "ifconfig",
+        encoding="ascii",
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        check=True,
+    )
+
+    eth_device = re.findall("eth[0-9]:.+\n.+[0-9]+", out.stdout)
+    inet = re.findall("inet [0-9.]+", str(eth_device))
+    netmask = re.findall("netmask [0-9.]+", str(eth_device))
+
+    host_setup = inet + netmask
+    return host_setup
+
+
+def en_setup(p):
+    """
+    This setup is used to create links with the ethernet interface using ifconfig and route.
+    It is strictly connected with the host IP because the setup will take it as a new IP for the
+    target with light modification.
+    up:      It indicates that you want this interface to be active.
+    wan:     It sets the default gateway using host parameters
+    ip:      It is determinate custom set ip for interface
+    dynamic: If True it will set ifconfig with dynamic flag, without netmask and ip
+    """
+
+    ifconfig_setup_cmd = "ifconfig en1 192.168.72.120 netmask 255.255.255.0 up"
+
+    psh.assert_prompt_after_cmd(p, ifconfig_setup_cmd, "success")
+
+
+def con_setup(dut: Dut, host_ip, host_port):
+    try:
+        host_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        host_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        host_socket.bind(('172.17.0.2', host_port))
```

Comment:
I'm thinking if you're using port forwarding through Docker, shouldn't you generally bind the socket in the container to something like `0.0.0.0:1025` or simply `:1025 `so that it listens on all interfaces inside the container, to enabling port forwarding?

Reply:
No, binding in this manner has nothing to do with port forwarding. I just used -p option in docker run command.